x-pack/plugins/observability/server/services/slo/tasks/summary_cleanup_task.ts

```diff
@@ -0,0 +1,235 @@
+import { SO_SLO_TYPE } from '../../../saved_objects';
+
+export const TASK_TYPE = 'SLO:SUMMARIES-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloSummaryIds: string[]) {
+  const missingSummaryIds = sloSummaryIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloSummaryIds.includes(item));
+
+  return {
+    missingSummaryIds,
+    missingSOIds,
+  };
+}
+
+const SEPARATOR = '__V__';
+
+export const getDeleteQueryFilter = (sloSummaryIdsToDelete: string[]) => {
+  return sloSummaryIdsToDelete.map((id) => {
+    const [sloId, revision] = id.split(SEPARATOR);
+    return {
+      bool: {
+        must: [
+          {
+            term: {
+              'slo.id': sloId,
+            },
+          },
+          {
+            term: {
+              'slo.revision': Number(revision),
+            },
+          },
+        ],
+      },
+    };
+  });
+};
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClientContract;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '3m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask();
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  runTask = async () => {
+    const runAt = new Date().toISOString();
+
+    if (this.soClient && this.esClient) {
+      const finder = this.soClient.createPointInTimeFinder<Pick<StoredSLO, 'id' | 'revision'>>({
+        type: SO_SLO_TYPE,
+        perPage: 1000,
+        fields: ['id', 'revision'],
+        namespaces: [ALL_SPACES_ID],
+      });
+      let searchAfterKey: SortResults | undefined;
+      let soIdsToCheck: string[] = [];
+      let sloSummaryIdsToDelete: string[] = [];
+
+      const processSloSummaries = async () => {
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
+        searchAfterKey = searchAfter;
+
+        const { missingSummaryIds } = findMissingItems(
+          soIdsToCheck,
+          sloSummaryIds.concat(sloSummaryIdsToDelete)
+        );
+
+        sloSummaryIdsToDelete = missingSummaryIds;
+      };
+
+      for await (const response of finder.find()) {
+        const soItems = response.saved_objects.map(
+          (so) => `${so.attributes.id}${SEPARATOR}${so.attributes.revision}`
+        );
+
+        soIdsToCheck = soIdsToCheck.concat(soItems);
```

Comment:
Do I understand correctly that we load all saved object "id__V__revision" into this array? 
If we are going to load all saved object in memory anyway, shouldn't we do that first, and then iterate on the processSloSummaries?